### ISSUE TITLE
Revise getting location logic

### ISSUE THREAD

**Issue by TimeString** [nesl/ActivityLabeling#3]
This is the only thing that I think is problematic so far, source:

https://github.com/nesl/ActivityLabeling/blob/master/app/src/main/java/com/ucla/zxxia/activitytracker/ActivityEditor.java#L83

In a nutshell, you use `getLastKnownLocation()` method, how do you know how long this location has been kept? It may be two hours ago and become irrevalent to the current location. The good thing is, as long as anybody requests a location, the location manager is going to keep a copy of this location instance, so one workaround is that you create a service whose job is solely getting the locations periodically (even don't need to store them). I think setting the interval as 1 minute is a good choice.

You also put the permission check in the function, which I would suggest moving this logic into another function which just does permission check. Also, usually permission check has to be done asynchronously. But one way to avoid this complication is to move the entire permission check to MainActivity.

In summary, you should do

- [x] Move the permission check to MainActivity
- [x] In my opinion, you can remove the logic which checks whether GPS is enabled or not, but if you want to keep it, also move it to MainActivity
- [x] Create a location service

In the end, I would expect `prepareLocation()` should be very should (only calls `getLastKnownLocation()`)

**Comment by TimeString** [comment]
P.S I thought we are going to manually enter the location??
